### PR TITLE
tests: Update tests to use specific resource versions

### PR DESCRIPTION
**Pull request by Harshil2107**
This update modifies the test configuration to specify the versions of resources used, rather than automatically using the latest versions. 
Previously, if a resource was updated for a change, it could potentially cause tests to fail if those tests were incompatible with the new version of the resource. 
Now, with this change, tests are tied to specific versions of resources, ensuring that any updates to resources will require corresponding updates to the tests to maintain compatibility.

Change-Id: I9633b1749f6c6c82af6aa6697b7e7656020f62fa

### CODE REVIEW

File: tests/gem5/checkpoint_tests/configs/x86-hello-restore-checkpoint.py [hunk 1]
```diff
@@ -62,7 +62,9 @@
 )
 board.set_se_binary_workload(
     obtain_resource("x86-hello64-static"),
```
Comment (powerjg) [inline tests/gem5/checkpoint_tests/configs/x86-hello-restore-checkpoint.py]: Why not add a version to this one?

File: tests/gem5/checkpoint_tests/configs/x86-fs-restore-checkpoint.py [hunk 2]
```diff
@@ -76,7 +76,9 @@
 board.set_kernel_disk_workload(
     kernel=obtain_resource("x86-linux-kernel-5.4.49"),
     disk_image=obtain_resource("x86-ubuntu-18.04-img"),
```
Comment (powerjg) [inline tests/gem5/checkpoint_tests/configs/x86-fs-restore-checkpoint.py]: Shouldn't these have versions too?

File: tests/gem5/checkpoint_tests/configs/power-hello-restore-checkpoint.py [hunk 3]
```diff
@@ -62,7 +62,9 @@
 )
 board.set_se_binary_workload(
     obtain_resource("power-hello"),
```
Comment (powerjg) [inline tests/gem5/checkpoint_tests/configs/power-hello-restore-checkpoint.py]: Shouldn't this have a version?

**Review by powerjg — COMMENTED** [review]
I pointed out some below, but there are many `obtain_resources` calls that don't have versions. I suggest in the **tests** we should have explicit versions everywhere.

**Review by ivanaamit — APPROVED** [review]
lgtm

**Pull request merged**